sigstore/dsse/__init__.py

```diff
@@ -228,6 +228,17 @@ def __eq__(self, other: object) -> bool:
 
         return self._inner == other._inner
 
+    @property
+    def signature(self) -> bytes:
+        """Return the decoded bytes of the Envelope signature."""
+        if len(self._inner.signatures) == 0:
+            return b""
+
+        signature_bytes = self._inner.signatures[0].sig
+        if not signature_bytes:
+            return b""
```

Comment:
Rather than forwarding these invalid states, we should enforce the invariant that a `dsse.Envelope` always has exactly one non-empty signature member.

This should be done in the `Envelope` constructor -- there's an example of this pattern here:

https://github.com/trail-of-forks/sigstore-python/blob/5f23327c1520b757670f6e3aeccda1da66c4805f/sigstore/models.py#L444

Reply:
To keep changes contained, this is part of an other PR : #1211